apps/system/js/base_module.js

```diff
@@ -111,6 +103,15 @@
   BaseModule.STATES = [];
 
   var SubmoduleMixin = {
+    loadWhenIdle: function(modules) {
+      return new Promise((resolve, reject) => {
+        this.service.request('schedule', () => {
+          this.constructor.SUB_MODULES =
+            this.constructor.SUB_MODULES.concat(modules);
+          this._startSubModules().then(resolve).catch(reject);
+        });
+      });
```

Comment:
Maybe this is equivalent to:

```javascript

return this.service.request('schedule', () => {
  this.constructor.SUB_MODULES =
    this.constructor.SUB_MODULES.concat(modules);
  return this._startSubModules();
});

```
?

The fact that `request` and `schedule` are both promises prevent me to understanding this clearly.